File: year2016/day03/python/solution.py
def main():
    print(part_1())
    print(part_2())


def part_1():
    return len([1 for a, b, c in input if check_triangle(a, b, c)])


def part_2():
    sum = 0
    for i in range(0, len(input), 3):
        for j in range(3):
            if check_triangle(input[i][j], input[i + 1][j], input[i + 2][j]):
                sum += 1

    return sum


def setup(path):
    global input
    
    with open(path, 'r') as file:
        input = [list(map(int, row.strip().split())) for row in file.read().split('\n')]


def check_triangle(a, b, c):
    return all([a + b > c, a + c > b, b + c > a])


if __name__ == '__main__':
    setup('../puzzle.txt')
    main()

File: year2016/day03/java/Solution.java
import java.nio.file.*;
import java.util.*;
import java.util.stream.*;


public class Solution {
  private static int[][] TRIANGLES;

  public static void main(String[] args) {
    try {
      setup(Files.readString(Paths.get("../puzzle.txt")));
      System.out.println(part1());
      System.out.println(part2());
    } catch(Exception e) {
      e.printStackTrace();
    }
  }

	public static int part1() {
    return (int) Arrays.stream(TRIANGLES).filter(triple -> checkTriangle(triple[0], triple[1], triple[2])).count();
  }

  public static int part2() {
    int sum = 0;
    for (int i = 0; i < TRIANGLES.length; i += 3) {
      for (int j = 0; j < 3; j++) {
        if (checkTriangle(TRIANGLES[i][j], TRIANGLES[i + 1][j], TRIANGLES[i + 2][j])) {
          sum++;
        }
      }
    }
    return sum;
  }

  public static void setup(String input) {
    TRIANGLES = Arrays.stream(input.split("\n"))
        .map(row -> Arrays.stream(row.trim().split("\\s+"))
        .mapToInt(Integer::parseInt)
        .toArray())
        .toArray(int[][]::new);
  }

  private static boolean checkTriangle(int a, int b, int c) {
    return Arrays.stream(new int[]{a + b, a + c, b + c}).allMatch(sum -> sum > (a + b + c - sum));
  }
}